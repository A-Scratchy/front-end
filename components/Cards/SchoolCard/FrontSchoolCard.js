import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Button from 'components/_common_/Button/Button';
import OutboundLink from 'components/_common_/OutboundLink/OutboundLink';
import DesktopIcon from 'static/images/icons/FontAwesome/desktop-solid.svg';
import DevicesIcon from 'static/images/icons/FontAwesome/devices-alt-solid.svg';
import HomeIcon from 'static/images/icons/FontAwesome/home-solid.svg';
import PeopleIcon from 'static/images/icons/FontAwesome/users-solid.svg';
import GIBillApprovedIcon from 'static/images/icons/gi-bill-approved.svg';
import GIBillUnavailableIcon from 'static/images/icons/gi-bill-unavailable.svg';
import Badge from 'components/Badge/Badge';
import styles from './FrontSchoolCard.css';

export default class FrontSchoolCard extends Component {
  static propTypes = {
    cardFlipCallback: PropTypes.func.isRequired,
    hasHardwareIncluded: PropTypes.bool.isRequired,
    hasHousing: PropTypes.bool.isRequired,
    hasOnline: PropTypes.bool.isRequired,
    hasOnlyOnline: PropTypes.bool.isRequired,
    isFullTime: PropTypes.bool.isRequired,
    locations: PropTypes.arrayOf(
      PropTypes.shape({
        city: PropTypes.string,
        va_accepted: PropTypes.bool.isRequired,
        state: PropTypes.string,
      }),
    ).isRequired,
    logoSource: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    website: PropTypes.string.isRequired,
  };

  render() {
    const { props } = this;

    return (
      <>
        <h5>{props.name}</h5>
        <img src={props.logoSource} alt="" />

        <div className={styles.interactionsContainer}>
          <Button>
            <OutboundLink analyticsEventLabel={`${props.name} | Website`} href={props.website}>
              Go To Website
            </OutboundLink>
          </Button>

          {props.locations.length > 1 ? (
            <Button
              analyticsObject={{
                action: 'Button Selected',
                category: 'Interactions',
                label: `${props.name} | Locations`,
              }}
              onClick={props.cardFlipCallback}
            >
              See Locations
            </Button>
          ) : (
            <span className={styles.singleLocationText}>
              {props.hasOnlyOnline
                ? 'Online Only'
                : `${props.locations[0].city}, ${props.locations[0].state}`}
            </span>
          )}
        </div>

        <hr className={styles.divider} />

        <div className={styles.detailsRow}>
          {/* GI Bill */}
          {props.locations.some(location => location.va_accepted) ? (
            <div className={`${styles.detailItem} ${styles.active}`}>
              <Badge
                label="G.I Bill"
                className={styles.codeSchoolCardIcon}
                icon={<GIBillApprovedIcon />}
              />
            </div>
          ) : (
            <div className={`${styles.detailItem} ${styles.disabled}`}>
              <Badge
                label="G.I Bill"
                className={styles.codeSchoolCardIcon}
                icon={<GIBillUnavailableIcon />}
              />
            </div>
          )}

          {/* Online Education */}
          <div
            className={classNames(styles.detailItem, {
              [styles.active]: props.hasOnline,
              [styles.disabled]: !props.hasOnline,
            })}
          >
            <Badge label="Online" className={styles.codeSchoolCardIcon} icon={<DesktopIcon />} />
          </div>

          {/* In-Person Education */}
          <div
            className={classNames(styles.detailItem, {
              [styles.active]: props.isFullTime,
              [styles.disabled]: !props.isFullTime,
            })}
          >
            {' '}
            <Badge label="In-Person" className={styles.codeSchoolCardIcon} icon={<PeopleIcon />} />
          </div>
        </div>

        <div className={styles.detailsRow}>
          {/* Equipment Provided */}
          <div
            className={classNames(styles.detailItem, {
              [styles.active]: props.hasHardwareIncluded,
              [styles.disabled]: !props.hasHardwareIncluded,
            })}
          >
            {' '}
            <Badge label="Equipment" className={styles.codeSchoolCardIcon} icon={<DevicesIcon />} />
          </div>

          {/* Housing Provided */}
          <div
            className={classNames(styles.detailItem, {
              [styles.active]: props.hasHousing,
              [styles.disabled]: !props.hasHousing,
            })}
          >
            {' '}
            <Badge label="Housing" className={styles.codeSchoolCardIcon} icon={<HomeIcon />} />
          </div>
        </div>
      </>
    );
  }
}
