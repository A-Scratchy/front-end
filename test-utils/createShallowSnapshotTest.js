/* eslint-env jest */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers or larger components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
