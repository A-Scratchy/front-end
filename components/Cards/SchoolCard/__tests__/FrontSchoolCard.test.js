import React from 'react';
import { mount } from 'enzyme';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';
import FrontSchoolCard from '../FrontSchoolCard';

const locations = [
  {
    va_accepted: true,
    address1: '825 Battery Street',
    address2: '3rd Floor',
    city: 'San Francisco',
    zip: '94111',
  },
  {
    va_accepted: true,
    address1: '825 Battery Street',
    address2: '3rd Floor',
    city: 'San Francisco',
    zip: '94111',
  },
];

describe('FrontSchoolCard', () => {
  it('should render with required props', () => {
    createShallowSnapshotTest(
      <FrontSchoolCard
        cardFlipCallback={() => {}}
        hasHardwareIncluded
        hasHousing
        hasOnline
        hasOnlyOnline={false}
        isFullTime
        locations={locations}
        logoSource="logoSource"
        name="school name"
        website="website"
      >
        <p>Testing!</p>
      </FrontSchoolCard>,
    );
  });

  it('should call callback function', () => {
    const onClickMock = jest.fn();
    const wrapper = mount(
      <FrontSchoolCard
        cardFlipCallback={onClickMock}
        hasHardwareIncluded
        hasHousing
        hasOnline
        hasOnlyOnline={false}
        isFullTime
        locations={locations}
        logoSource="logoSource"
        name="school name"
        website="website"
      />,
    );
    wrapper
      .find('button')
      .filterWhere(node => node.text() === 'See Locations')
      .simulate('click');
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
