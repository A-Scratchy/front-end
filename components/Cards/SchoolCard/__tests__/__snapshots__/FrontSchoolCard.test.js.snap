// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontSchoolCard should render with required props 1`] = `
<React.Fragment>
  <h5>
    school name
  </h5>
  <img
    alt=""
    src="logoSource"
  />
  <div
    className="interactionsContainer"
  >
    <Button
      analyticsObject={
        Object {
          "action": "Button Selected",
          "category": "Interactions",
        }
      }
      datum=""
      disabled={false}
      fullWidth={false}
      onClick={[Function]}
      tabIndex={0}
      theme="primary"
      type="button"
    >
      <OutboundLink
        analyticsEventLabel="school name | Website"
        hasIcon={true}
        href="website"
      >
        Go To Website
      </OutboundLink>
    </Button>
    <Button
      analyticsObject={
        Object {
          "action": "Button Selected",
          "category": "Interactions",
          "label": "school name | Locations",
        }
      }
      datum=""
      disabled={false}
      fullWidth={false}
      onClick={[Function]}
      tabIndex={0}
      theme="primary"
      type="button"
    >
      See Locations
    </Button>
  </div>
  <hr
    className="divider"
  />
  <div
    className="detailsRow"
  >
    <div
      className="detailItem active"
    >
      <Badge
        className="codeSchoolCardIcon"
        icon={<Svg />}
        isImageFirst={true}
        label="G.I Bill"
      />
    </div>
    <div
      className="detailItem active"
    >
      <Badge
        className="codeSchoolCardIcon"
        icon={<Svg />}
        isImageFirst={true}
        label="Online"
      />
    </div>
    <div
      className="detailItem active"
    >
      <Badge
        className="codeSchoolCardIcon"
        icon={<Svg />}
        isImageFirst={true}
        label="In-Person"
      />
    </div>
  </div>
  <div
    className="detailsRow"
  >
    <div
      className="detailItem active"
    >
      <Badge
        className="codeSchoolCardIcon"
        icon={<Svg />}
        isImageFirst={true}
        label="Equipment"
      />
    </div>
    <div
      className="detailItem active"
    >
      <Badge
        className="codeSchoolCardIcon"
        icon={<Svg />}
        isImageFirst={true}
        label="Housing"
      />
    </div>
  </div>
</React.Fragment>
`;
